### PR TITLE
Update index.ts

### PR DESCRIPTION
**Pull request by JOnMila**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,37 +1,152 @@
...
+  const result = await prisma.user.create({
+    data: {
+      name,
+      email,
+      posts: {
+        create: postData,
+      },
+    },
+  })
+  res.json(result)
```
Comment (adasarpan404) [inline index.ts]: add statuscode here

File: index.ts [hunk 2]
```diff
@@ -1,37 +1,152 @@
...
+app.post(`/post`, async (req, res) => {
+  const { title, content, authorEmail } = req.body
+  const result = await prisma.post.create({
+    data: {
+      title,
+      content,
+      author: { connect: { email: authorEmail } },
+    },
+  })
+  res.json(result)
```
Comment (adasarpan404) [inline index.ts]: same here

File: index.ts [hunk 3]
```diff
@@ -1,37 +1,152 @@
...
         },
+      },
     })
 
-    const allUsers = await prisma.user.findMany({
-        include: {
-            posts: true,
-            profile: true,
-        },
+    res.json(post)
```
Comment (adasarpan404) [inline index.ts]: same goes here

File: index.ts [hunk 4]
```diff
@@ -1,37 +1,152 @@
...
+  } catch (error) {
+    res.json({ error: `Post with ID ${id} does not exist in the database` })
```
Comment (adasarpan404) [inline index.ts]: also add status code here

File: index.ts [hunk 5]
```diff
@@ -1,37 +1,152 @@
...
+      },
+    })
+
+    const updatedPost = await prisma.post.update({
+      where: { id: Number(id) || undefined },
+      data: { published: !postData?.published },
     })
-    console.dir(allUsers, { depth: null })
-}
+    res.json(updatedPost)
```
Comment (adasarpan404) [inline index.ts]: also add status code 200

File: index.ts [hunk 6]
```diff
@@ -1,37 +1,152 @@
...
+  } catch (error) {
+    res.json({ error: `Post with ID ${id} does not exist in the database` })
```
Comment (adasarpan404) [inline index.ts]: also add status code 500

**Review by adasarpan404 — CHANGES_REQUESTED** [review]
update my review

File: index.ts (hunk 2 above)
Comment (JOnMila) [reply index.ts]: done

File: index.ts [hunk 7]
```diff
@@ -1,37 +1,152 @@
...
     })
 
-    const allUsers = await prisma.user.findMany({
-        include: {
-            posts: true,
-            profile: true,
-        },
+    res.status(200).json(post)
+  } catch (error) {
+    res.json({ error: `Post with ID ${id} does not exist in the database` })
```
Comment (adasarpan404) [inline index.ts]: you have forgotten here

File: index.ts [hunk 8]
```diff
@@ -1,37 +1,152 @@
...
+
+    const updatedPost = await prisma.post.update({
+      where: { id: Number(id) || undefined },
+      data: { published: !postData?.published },
     })
-    console.dir(allUsers, { depth: null })
-}
+    res.status(200).json(updatedPost)
+  } catch (error) {
+    res.json({ error: `Post with ID ${id} does not exist in the database` })
```
Comment (adasarpan404) [inline index.ts]: same you forgotten here

**Review by adasarpan404 — CHANGES_REQUESTED** [review]

File: index.ts (hunk 7 above)
Comment (JOnMila) [reply index.ts]: completed

File: index.ts (hunk 8 above)
Comment (JOnMila) [reply index.ts]: completed

**Review by adasarpan404 — APPROVED** [review]
LGTM

**Pull request merged**